### PR TITLE
cancel button implemented, need to be some testing

### PR DESCRIPTION
**Pull request by alaapbharadwaj**
## Description

Cancel button redirects and don't show again checkbox is working as well

## Tickets

* [CNFT1-2997](https://cdc-nbs.atlassian.net/browse/CNFT1-2997)

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT1-2997]: https://cdc-nbs.atlassian.net/browse/CNFT1-2997?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx [hunk 1]
```diff
@@ -32,6 +40,28 @@ export const AddPatientExtended = () => {
...
+
+    const handleConfirmCancel = () => {
+        save(visibilityCheckBox);
+        closeForm();
+    };
+
+    const closeForm = () => {
+        form.reset();
+        history.go(-1);
+    };
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx:63]: Question: should clicking cancel be the same as pushing the back button or should it have defined behavior (go to a certain route)

Comment (alaapbharadwaj) [reply apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx:63]: For my understanding, you can only go to extended form through the basic form, and the expected behavior is to go back to the basic form. Should I be more intentional with the redirect?

Comment (adamloup-enquizit) [reply apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx:63]: You could go directly to the URL, is the `form.reset` requried?

File: apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx [hunk 2]
```diff
@@ -11,12 +11,20 @@ import { AddExtendedPatientInteractionProvider } from './useAddExtendedPatientIn
...
+import { Checkbox } from '@trussworks/react-uswds';
+import { useLocalStorage } from 'storage';
 import { Shown } from 'conditional-render';
 import { PatientCreatedPanel } from '../PatientCreatedPanel';
 import { CreatedPatient } from './api';
 
 export const AddPatientExtended = () => {
     const interaction = useAddExtendedPatient({ transformer, creator });
+    const [cancelModal, setCancelModal] = useState<boolean>(false);
+    const [visibilityCheckBox, setVisibilityCheckBox] = useState<boolean>(false);
```
Comment (adamloup-enquizit) [inline apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx:26]: suggestion:   The `visibilityCheckBox` state is only used by the confirmation modal, you could create a `CancelAddPatientExtendedPanel` component that could encapsulate the logic for cancel

Comment (adamloup-enquizit) [reply apps/modernization-ui/src/apps/patient/add/extended/AddPatientExtended.tsx:26]: suggestion: The name `visibilityCheckBox` doesn't really express what that actually is, it's more `doNotShowAgain`

**Review by mpeels — DISMISSED** [review]

**Review by stevegsa — DISMISSED** [review]

**Review by mpeels — APPROVED** [review]

**Review by adamloup-enquizit — APPROVED** [review]
👍

**Pull request merged**